test/src/org/labkey/test/tests/targetedms/TargetedMSProteinSequenceViewTest.java

```diff
@@ -0,0 +1,154 @@
+        waitAndClickAndWait(Locator.linkWithText(CS_SKY_FILE));
+        clickAndWait(Locator.linkWithText("sp|O13527|YA11B_YEAST"));
+
+        log("Verifying Intensity values");
+        SequenceCoverageWebPart sequenceCoverage = new SequenceCoverageWebPart(getDriver());
+        checker().verifyEquals("Incorrect default selection of display By", "Intensity", sequenceCoverage.getDisplayBy());
+        checker().verifyEquals("Incorrect heatMap legend values for Intensity", Arrays.asList("6.85", "5.27", "3.68"), sequenceCoverage.getHeatMapLegendValues());
+        checker().verifyEquals("Incorrect Peptide Details for Intensity 23(Blue)", "Mass: 2890.21\n" +
+                "Start: 99\n" +
+                "End: 123\n" +
+                "Unmodified: 1\n" +
+                "Intensity Rank: 23\n" +
+                "Raw Intensity: 4.821E+03\n" +
+                "Log 10 Base Intensity: 3.68", sequenceCoverage.getPopUpDetails("23"));
+        checker().verifyEquals("Incorrect peptide details for Intensity 4(Red)", "Mass: 1675.81\n" +
+                "Start: 123\n" +
+                "End: 138\n" +
+                "Unmodified: 1\n" +
+                "Intensity Rank: 4\n" +
+                "Raw Intensity: 4.753E+06\n" +
+                "Log 10 Base Intensity: 6.68", sequenceCoverage.getPopUpDetails("4"));
+        checker().screenShotIfNewError("Intensity_Errors");
+
+        log("Verifying replicate selection with intensity");
+        checker().verifyEquals("Incorrect default selection for Replicate", "All", sequenceCoverage.getReplicate());
+        sequenceCoverage.setReplicate("20170901_CalCurves_YeastCurve_A_1ug-ul_BY4741_166");
+        sequenceCoverage = new SequenceCoverageWebPart(getDriver());
+        checker().verifyEquals("Incorrect heatMap legend values for Intensity and single replicate", Arrays.asList("6.66", "5.17", "3.68"), sequenceCoverage.getHeatMapLegendValues());
+        checker().verifyEquals("Incorrect Peptide Details for Intensity 15(Blue) and single replicate", "Mass: 2890.21\n" +
+                "Start: 99\n" +
+                "End: 123\n" +
+                "Unmodified: 1\n" +
+                "Intensity Rank: 15\n" +
+                "Raw Intensity: 4.821E+03\n" +
+                "Log 10 Base Intensity: 3.68", sequenceCoverage.getPopUpDetails("15"));
+        checker().verifyEquals("Incorrect peptide details for Intensity 1(Red) and single replicate", "Mass: 816.93\n" +
+                "Start: 218\n" +
+                "End: 225\n" +
+                "Unmodified: 1", sequenceCoverage.getPopUpDetails("1"));
+        checker().screenShotIfNewError("Intensity_And_SingleReplicate_Errors");
+
+        log("Verifying Confidence score value");
+        sequenceCoverage.setDisplayBy("Confidence Score");
+        sequenceCoverage = new SequenceCoverageWebPart(getDriver());
+        sequenceCoverage.setReplicate("All");
+        sequenceCoverage = new SequenceCoverageWebPart(getDriver());
```

Comment:
`SequenceCoverageWebPart` should be able to handle these operations without getting into a bad state. The standard pattern is to wait for the update and then clear the element cache. `TimeChartWizard.doAndWaitForUpdate` has an implementation that's probably similar to what this webpart might need.

Reply:
done